diff --git a/kernel/sucompat.c b/kernel/sucompat.c
index 04d8d3ca..patched 100644
--- a/kernel/sucompat.c
+++ b/kernel/sucompat.c
@@ -11,6 +11,12 @@
 #include <linux/version.h>
 #include <linux/sched/task_stack.h>
 #include <linux/ptrace.h>
+#ifdef CONFIG_KSU_SUSFS
+#include <linux/susfs_def.h>
+#include <linux/namei.h>
+#include "selinux/selinux.h"
+#include "objsec.h"
+#endif
 
 #include "allowlist.h"
 #include "feature.h"
