diff --git a/kernel/kernel_umount.c b/kernel/kernel_umount.c
index a58d2f75..f6fc3cc8 100644
--- a/kernel/kernel_umount.c
+++ b/kernel/kernel_umount.c
@@ -107,6 +107,7 @@ int ksu_handle_umount(uid_t old_uid, uid_t new_uid)
         return 0;
     }

+#ifndef CONFIG_KSU_SUSFS
     // There are 5 scenarios:
     // 1. Normal app: zygote -> appuid
     // 2. Isolated process forked from zygote: zygote -> isolated_process
@@ -130,6 +131,8 @@ int ksu_handle_umount(uid_t old_uid, uid_t new_uid)
         pr_info("handle umount ignore non zygote child: %d\n", current->pid);
         return 0;
     }
+#endif // #ifndef CONFIG_KSU_SUSFS
+
     // umount the target mnt
     pr_info("handle umount for uid: %d, pid: %d\n", new_uid, current->pid);
