diff --git a/kernel/supercalls.c b/kernel/supercalls.c
index 0a9fc63d..patched 100644
--- a/kernel/supercalls.c
+++ b/kernel/supercalls.c
@@ -14,6 +14,10 @@
 #include <linux/task_work.h>
 #include <linux/uaccess.h>
 #include <linux/version.h>
+#ifdef CONFIG_KSU_SUSFS
+#include <linux/namei.h>
+#include <linux/susfs.h>
+#endif
 
 #include "supercalls.h"
 #include "arch.h"
@@ -28,6 +32,10 @@
 #include "file_wrapper.h"
 #include "syscall_hook_manager.h"
 
+#ifdef CONFIG_KSU_SUSFS
+bool susfs_is_boot_completed_triggered __read_mostly = false;
+#endif
+
 // Permission check functions
 bool only_manager(void)
 {
@@ -108,6 +116,9 @@ static int do_report_event(void __user *arg)
 		boot_complete_lock = true;
 		pr_info("boot_complete triggered\n");
 		on_boot_completed();
+#ifdef CONFIG_KSU_SUSFS_SUS_MOUNT
+		susfs_is_boot_completed_triggered = true;
+#endif
 	}
 	break;
 }
