--- a/kernel/ksu.c
+++ b/kernel/ksu.c
@@ -3,6 +3,9 @@
 #include <linux/kobject.h>
 #include <linux/module.h>
 #include <linux/workqueue.h>
+#ifdef CONFIG_KSU_SUSFS
+#include <linux/susfs.h>
+#endif
 
 #include "allowlist.h"
 #include "feature.h"
@@ -36,6 +39,10 @@ int __init kernelsu_init(void)
 
 	ksu_supercalls_init();
 
+#ifdef CONFIG_KSU_SUSFS
+	susfs_init();
+#endif
+
 	ksu_syscall_hook_manager_init();
 
 	ksu_allowlist_init();
