--- a/kernel/supercalls.c
+++ b/kernel/supercalls.c
@@ -11,6 +11,10 @@
 #include <linux/task_work.h>
 #include <linux/uaccess.h>
 #include <linux/version.h>
+#ifdef CONFIG_KSU_SUSFS
+#include <linux/namei.h>
+#include <linux/susfs.h>
+#endif
 
 #include "arch.h"
 #include "allowlist.h"
@@ -25,6 +29,10 @@
 #include "file_wrapper.h"
 #include "syscall_hook_manager.h"
 
+#ifdef CONFIG_KSU_SUSFS
+bool susfs_is_boot_completed_triggered __read_mostly = false;
+#endif
+
 // Permission check functions
 bool only_manager(void)
 {
@@ -107,6 +115,9 @@ static int do_report_event(void __user *arg)
 		boot_complete_lock = true;
 		pr_info("boot_complete triggered\n");
 		on_boot_completed();
+#ifdef CONFIG_KSU_SUSFS_SUS_MOUNT
+		susfs_is_boot_completed_triggered = true;
+#endif
 	}
 	break;
 }
